options.go

```diff
@@ -587,3 +588,25 @@ func DialRanker(d network.DialRanker) Option {
 		return nil
 	}
 }
+
+// UDPBlackHoleConfig configures libp2p to use c as the config for UDP black hole detection
+func UDPBlackHoleConfig(c *swarm.BlackHoleConfig) Option {
+	return func(cfg *Config) error {
+		if c == nil {
+			return errors.New("udp black hole config cannot be nil")
+		}
+		cfg.UDPBlackHoleConfig = c
+		return nil
+	}
+}
+
+// IPv6BlackHoleConfig configures libp2p to use c as the config for IPv6 black hole detection
+func IPv6BlackHoleConfig(c *swarm.BlackHoleConfig) Option {
+	return func(cfg *Config) error {
+		if c == nil {
+			return errors.New("ipv6 black hole config cannot be nil")
+		}
+		cfg.IPv6BlackHoleConfig = c
+		return nil
+	}
+}
```

Comment:
What about making this more general by having a `SwarmOpts(opts ...swarm.Option) Option` here?

Reply:
Your suggestion is right. I did it this way because none of the other swarm.Options were exposed here. We should also fix the other cases like this one https://github.com/libp2p/go-libp2p/blob/master/config/config.go#L176